### PR TITLE
Feature/agent profiles

### PR DESCRIPTION
**Pull request by eric-gt**
- /pages/api/xAPI/agents/profiles endpoint
- /pages/api/xAPI/statements endpoint
- postgresql database with Prisma ORM
- all tests passing

### CODE REVIEW

File: test/pages/api/xAPI/statements/index.spec.ts [hunk 1]
```diff
@@ -12,7 +12,7 @@ const pw = "test-password";
 
 beforeEach(async () => {
   // reset db
-  await dbClient.agentAccount.deleteMany({});
+  // await dbClient.agentAccount.deleteMany({});
```
Comment (calufa) [inline test/pages/api/xAPI/statements/index.spec.ts:15]: If we don't need it, why not remove it entirely?

File: models/agent.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import dbClient from "../lib/db";
+
+export async function createAgent(agent) {
+  const id = await dbClient.agent.create({data: agent, select: id});
```
Comment (calufa) [inline models/agent.ts:4]: > `select: id`, 

where does `id` come from?

File: models/agent.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  return id;
+}
+
+export async function findByAgent(agent) {
+    const agentId = await dbClient.agent.findFirst({
+     where: {
+       mbox: agent.mbox? agent.mbox : null,
+       mbox_sha1sum: agent.mbox_sha1sum? agent.mbox_sha1sum : null,
+       openid: agent.openId? agent.openid : null
```
Comment (calufa) [inline models/agent.ts:13]: Shouldn't we have a comma (`,`) at the end of the line? Strangely, the test suite is not breaking... hmm...

File: models/agentProfile.ts [hunk 4]
```diff
@@ -0,0 +1,104 @@
+import dbClient from "../lib/db";
+
+export async function add(profiles) {
+  if(!Array.isArray(profiles)) {
+    profiles = [profiles]
+  }
```
Comment (calufa) [inline models/agentProfile.ts:6]: I don't really like the idea of methods checking and modifying inputs and _accepting_ different types (string and array) for the same param. IMO, it makes the API contract/interface more complex to understand for the user.

Have you considered:
1. Doing the checking and modification of `profiles` into an array before calling `add()`. And always send an array.
2. Replacing `add()` for two methods `addProfile()` and `addProfiles()`. Where `addProfile()` builds an array and then calls `addProfiles()`.

File: models/agentProfile.ts [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+    }
+  });
+
+  return rows;
+}
+
+export async function getProfile(agent, profileId) {
+  const row = await dbClient.agentProfile.findFirst({
+    where: {
+      agent: agent,
```
Comment (calufa) [inline models/agentProfile.ts:72]: One unique feature of ES6 the "Object Property Value Shorthand".

"Object Property Value Shorthand" allows you to represent this:
```
where: {
  agent: agent,
  ...
}
```

as

```
where: {
  agent,
  ...
}
```

https://alligator.io/js/object-property-shorthand-es6/

Comment (calufa) [reply models/agentProfile.ts:72]: It is mostly a cosmetic suggestion, as the code will run the same.

File: test/pages/api/xAPI/agents/profile/index.spec.ts [hunk 6]
```diff
@@ -0,0 +1,230 @@
...
+import dbClient from "../../../../../../lib/db";
+import handlers from "../../../../../../pages/api/xAPI/agents/profile/index";
+import { testServer } from "../../../../../server";
+
+const usr = "test-user";
+const pw= "test-password";
+
+beforeEach(async() => {
+  //reset db
+  // await dbClient.agentAccount.deleteMany({});
```
Comment (calufa) [inline test/pages/api/xAPI/agents/profile/index.spec.ts:16]: If we don't need it, why not remove it entirely?

File: test/pages/api/xAPI/agents/profile/index.spec.ts [hunk 7]
```diff
@@ -0,0 +1,230 @@
...
+      .send(profileCollection.profiles)
+      .expect(204);
+  });
+  it("returns 204 No Content", async () => {
+    await testServer(handlers)
+      .post("/")
+      .auth(usr, pw)
+      .send(singleAgentProfile.profile)
+      .expect(204);
+  });
```
Comment (calufa) [inline test/pages/api/xAPI/agents/profile/index.spec.ts:40]: This code looks repeated, see lines 27-33.

File: pages/api/xAPI/agents/profile/index.ts [hunk 8]
```diff
@@ -0,0 +1,87 @@
+import Cors from "cors";
+import deepEqual from "deep-equal";
+import type { NextRequest, NextResponse } from "next/server";
+import { apiHandler } from "../../../helpers/api/api-handler";
+import middleware, { Next } from "../../../helpers/api/request-sanitizers";
+import dbClient from "../../../../../lib/db";
```
Comment (calufa) [inline pages/api/xAPI/agents/profile/index.ts:6]: It doesn't seem like we are using `Cors`, `deepEqual`, `NextRequest`, `NextResponse`, `Next`,  or `dbClient`.

**Comment by calufa** [comment]
Looks good! Just left a few comments.

File: models/agent.ts (hunk 2 above)
Comment (eric-gt) [reply models/agent.ts:4]: `id` is an internal-to-postgres field that Prisma requires in order to represent the AgentProfile-to-Agent relation
https://www.prisma.io/docs/concepts/components/prisma-schema/relations#the-relation-attribute
you can see how it connects in `schema.prisma`
```
model AgentProfile {
  ...
  agent      Agent       @relation(references: [id], fields: [agentId])
  agentId    String
...
}

model Agent {
  id                         String @default(cuid())
....
  agentProfiles      AgentProfile[]
}
```

File: models/agentProfile.ts (hunk 4 above)
Comment (eric-gt) [reply models/agentProfile.ts]: good idea! I went with option 1

**Pull request merged**